contracts/governance/StkTruToken.sol

```diff
@@ -0,0 +1,89 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.6.10;
+
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {ReentrancyGuard} from "@openzeppelin/contracts/utils/ReentrancyGuard.sol";
+import {SafeMath} from "@openzeppelin/contracts/math/SafeMath.sol";
+
+import {VoteToken} from "./VoteToken.sol";
+import {ClaimableContract} from "../trusttoken/common/ClaimableContract.sol";
+import {ITruPriceOracle} from "./interface/ITruPriceOracle.sol";
+
+contract StkTruToken is VoteToken, ClaimableContract, ReentrancyGuard {
+    using SafeMath for uint256;
+
+    uint256 public constant MIN_STAKE_TIME = 14 days;
+
+    IERC20 public trustToken;
+    IERC20 public tusd;
+    ITruPriceOracle public oracle;
+    mapping(address => uint256) public unlockTime;
+
+    event OracleChanged(ITruPriceOracle newOracle);
+    event Stake(address indexed staker, uint256 amount, uint256 minted);
+    event Unstake(address indexed staker, uint256 burntAmount, uint256 truAmount, uint256 tusdAmount);
+
+    function initialize(
+        IERC20 _trustToken,
+        IERC20 _tusd,
+        ITruPriceOracle _oracle
+    ) public {
+        require(!initalized, "StkTruToken: Already initialized");
+        trustToken = _trustToken;
+        tusd = _tusd;
+        oracle = _oracle;
+        owner_ = msg.sender;
+        initalized = true;
+    }
+
+    function setOracle(ITruPriceOracle _oracle) external onlyOwner {
+        oracle = _oracle;
+        emit OracleChanged(_oracle);
+    }
+
+    function stake(uint256 amount) external {
+        unlockTime[msg.sender] = block.timestamp + MIN_STAKE_TIME;
```

Comment:
rather than storing the unlock time, we want to have a function to initiate cooldown. Initiating cooldown will set unlcokTime[msg.sender] to block.timestamp + MIN_STAKE_TIME. Staking more should increase the cooldown. We don't have a definite formula for how to increase the cooldown, so It think it's fine to just reset the cooldown to max if someone stakes more after initiating cooldown.

See stkAave: https://etherscan.io/address/0x4da27a545c0c5b758a6ba100e3a049001de870f5#code